gobblin-runtime/src/main/java/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -66,6 +72,14 @@
 
   private final Path outputTaskStateDir;
 
+  /**
+   * Add a cloesable action to run after each existence-checking of task state file.
+   * A typical example to plug here is hive registration:
+   * We do hive registration everytime there are available taskStates deserialized from storage, on the driver level.
+   */
+  public final Optional<Closeable> optionalTaskCollectorHandler;
+  private final Closer handlerCloser = Closer.create();
```

Comment:
Doesn't appear to be used. Did you forget to register the handler to the closer?